Load uri with auth

keis opened the pull request:
Still need to tests this but this shows the basic idea.

I went with changing %r binding to not add quotes, another option was setting the split option for uri but I think that's a bit more risky.

mathstuf commented on examples/data/scripts/history.sh:
```diff
@@ -7,4 +7,8 @@
 
-print "$( date +'%Y-%m-%d %H:%M:%S' ) $UZBL_URI $UZBL_TITLE\n" >> "$UZBL_HISTORY_FILE"
+stripauth () {
+    print $@ | sed 's;\([a-z]*\)://.*@;\1://;'
```
Use `-e` for `sed` and quote the `print` arguments.

keis replied:
Fixed

mathstuf commented on uzbl/plugins/cmd_expand.py:
```diff
@@ -10,7 +10,7 @@ def cmd_expand(cmd, args):
 
         %s = replace('%s', ' '.join(args))
-        %r = replace('%r', "'%s'" % escaped(' '.join(args)))
+        %r = replace('%r', "%s" % escaped(' '.join(args)))
```
The docs probably need updating for this change.

keis replied:
Yes, good point.

keis commented:
In order to get useful integration tests we would need to have the em bit live and working for the tests because so much is back-and-forth between core and em to do anything.

I had this idea of writing these tests in python and using a instrumented em to log events and to ensure the minimum required set of plugins is loaded for each scenario. #268 moves a bit in this direction by making the main class a bit reusable but there's a lot of work involved to actually get it to fly though.

mathstuf commented on README.event-manager.md:
```diff
@@ -103,7 +103,9 @@ may also be an empty string.
-* `%r`: The string, escaped and quoted for uzbl.
+* `%r`: The string, escaped for uzbl.
+  This format used to also wrap the string in single-quotes. The same result
+  can still be achived by include the quotes in the format string (e.g `'%r'`)
```
There are lots of `%r` instances in the config that probably need updated (primarily the uses in search bindings).

mathstuf replied:
Hrm. Maybe it's just easier to introduce a new `%` specifier…

keis replied:
Another option would be to set SPLIT of the uri command so that commands like `uri 'http://uzbl.org'` works. But I think that is perhaps riskier.

keis replied:
but OTOH that would only affect the uri-command and nothing else and in general it's preferable if commands interpret quoted strings as expected

keis replied:
yep. that looks to be it. `include` would also be trivial to change to use split.

I think either way it's not a big worry for porting as it's not a fundamental change but just some escaping noise. I'll can take a look and see how much would break from changing uri.

mathstuf commented:
Is the sandbox environment not sufficient? It sets up the default configuration and such, but we can use the `expect` tool to run a test by connecting to the socket, right? The sandbox could probably be expanded to also use a special config file for more specific testing.

keis commented:
Not familiar with the expect tool but it looks like something like that could work. The sandbox should be fine and would be required either way for the supporting shell scripts etc.

keis commented:
Updated. changed `uri` and I think that turned out better. Also noticed that the current escape of %r is not very useful if trying to pass through strings with quotes in them. Maybe I'm missing some bit of history here but to me it looks like the old behaviour was simply broken.

mathstuf commented:
Docs were dropped, but other than that, much cleaner :) .

keis commented:
Ah yes the docs went with the commit I dropped. Need to rewrite that anyway to match the new changes.

keis commented:
Added a small docs update but I'm not really sure what the old rules where supposed to solve so it's perhaps not super useful.
